components/witness/src/tests.rs

```diff
@@ -425,84 +425,93 @@ fn test_mbx() {
+            )
             .unwrap();
-        std::fs::create_dir_all(root_witness.path()).unwrap();
-        Witness::new(root_witness.path(), signer.clone().public_key()).unwrap()
-    };
 
-    let inception_event = controller
-        .incept(
-            Some(vec![witness.prefix.clone()]),
-            Some(SignatureThreshold::Simple(1)),
-        )
-        .unwrap();
+        // send to witness
+        witness
+            .process(&vec![Message::Event(incept_event.clone())])
+            .unwrap();
+        witness.respond(signer.clone()).unwrap();
+    }
 
-    witness
-        .process(&vec![Message::Event(inception_event.clone())])
-        .unwrap();
-    witness.respond(signer.clone()).unwrap();
-
-    let qry_msg = QueryEvent::new_query(
-        QueryRoute::Mbx {
-            args: QueryArgsMbx {
-                i: IdentifierPrefix::Basic(witness.prefix.clone()),
-                pre: controller.prefix().clone(),
-                src: IdentifierPrefix::Basic(witness.prefix.clone()),
-                topics: QueryTopics {
-                    credential: 0,
-                    receipt: 0,
-                    replay: 0,
-                    multisig: 0,
-                    delegate: 0,
+    // query witness
+    for controller in controllers {
+        let qry_msg = QueryEvent::new_query(
+            QueryRoute::Mbx {
+                args: QueryArgsMbx {
+                    i: IdentifierPrefix::Basic(witness.prefix.clone()),
+                    pre: controller.prefix().clone(),
+                    src: IdentifierPrefix::Basic(witness.prefix.clone()),
+                    topics: QueryTopics {
+                        credential: 0,
+                        receipt: 0,
+                        replay: 0,
+                        multisig: 0,
+                        delegate: 0,
+                    },
                 },
+                reply_route: "".to_string(),
             },
-            reply_route: "".to_string(),
-        },
-        SerializationFormats::JSON,
-        &SelfAddressing::Blake3_256,
-    )
-    .unwrap();
-
-    let signature = signer.sign(qry_msg.serialize().unwrap()).unwrap();
-    let signatures = vec![AttachedSignaturePrefix::new(
-        SelfSigning::Ed25519Sha512,
-        signature,
-        0,
-    )];
-
-    let mbx_msg = Message::Query(SignedQuery::new(
-        qry_msg,
-        controller.prefix().clone().clone(),
-        signatures,
-    ));
-
-    witness.process(&[mbx_msg]).unwrap();
-    let receipts = &witness.respond(signer.clone()).unwrap();
-
-    assert_eq!(receipts.len(), 1);
-    let receipt = receipts[0].clone();
+            SerializationFormats::JSON,
+            &SelfAddressing::Blake3_256,
+        )
+        .unwrap();
 
-    let receipt = if let Message::NontransferableRct(receipt) = receipt {
-        receipt
-    } else {
-        panic!("didn't receive a receipt")
-    };
+        let signature = signer.sign(qry_msg.serialize().unwrap()).unwrap();
```

Comment:
```suggestion
        let signature = controller.key_manager().lock().unwrap().sign(&qry_msg.serialize().unwrap()).unwrap();
```
`qry` should be signed by controller, not by witness.